massgov/mayflower
# Assets/multilingual font compression

## Description
-This PR fixes bugs in the loading of the following fonts that were already in assets but not displaying properly:
--Hebrew
--Myanmar
--Greek
--Cyrillic
-Re-factors Latin by adding an extra font to cover Latin characters used by other languages like Vietnamese
-Adds the following fonts supported by Mass.gov that were not in assets yet:
--Armenian
--Devanagari
--Gujarati
--Khmer
--Lao
--Malayalam
--Tamil
--Telugu
--Thai
-Updates the readme in fonts with the python scripts for subsetting fonts
-Updates the typography page in core 

## Related Issue / Ticket
- [JIRA issue - DP-20436]
- [JIRA issue - DP-20682]
- [JIRA issue - DP-20686]

## Steps to Test
1. Pull the branch and run rush start:core
2. Open firefox and go to http://localhost:6006/?path=/docs/foundation-typography--page
3. look at the  screenshots which provide some insight on how you can highlight the specific characters loading in a font using the FF inspectors fonts panel
4. Open the inspector and the fonts tab and check that all of the fonts are loading

## Screenshots
### From Core
<img width="1118" alt="Screen Shot 2020-12-23 at 4 06 58 PM" src="https://user-images.githubusercontent.com/3894652/103041275-8fa0d000-453b-11eb-8999-a9acc53f0777.png">
<img width="1108" alt="Screen Shot 2020-12-23 at 4 07 17 PM" src="https://user-images.githubusercontent.com/3894652/103041277-92032a00-453b-11eb-86c0-b685c413c408.png">
<img width="1139" alt="Screen Shot 2020-12-23 at 4 07 32 PM" src="https://user-images.githubusercontent.com/3894652/103041279-93345700-453b-11eb-9d7d-7e37a88e7d4a.png">
<img width="1080" alt="Screen Shot 2020-12-23 at 4 07 43 PM" src="https://user-images.githubusercontent.com/3894652/103041282-94658400-453b-11eb-9a3d-b12a2ef028d3.png">
<img width="1157" alt="Screen Shot 2020-12-23 at 4 09 42 PM" src="https://user-images.githubusercontent.com/3894652/103041283-94fe1a80-453b-11eb-99b8-1d66b3e9cda4.png">
<img width="1159" alt="Screen Shot 2020-12-23 at 4 09 51 PM" src="https://user-images.githubusercontent.com/3894652/103041286-97607480-453b-11eb-8351-2cac2ddef5a0.png">
<img width="1143" alt="Screen Shot 2020-12-23 at 4 11 07 PM" src="https://user-images.githubusercontent.com/3894652/103041289-9891a180-453b-11eb-856f-eab1072f7f3d.png">

#### @TODO
-Gujarati has no VF
-All of the newly added fonts still need eot fallbacks
-There are no py scripts in the readme yet for Gujarati, Devanagari, Tamil, Telugu, and Thai
-CJK fonts  (Chinese Japanese, Korean) are still not done
-Mono has no eot fallbacks yet
-The latin-ext VF does not have non-variable fallbacks or eot and the woff2 has a bug so it's commented out, but the woff works.
